tests/testsuite.src/used_binaries.at

```diff
@@ -396,11 +396,16 @@ AT_DATA([prog.cob], [
            EXIT PROGRAM.
 ])
 
-AT_CHECK([TMPDIR="" TMP="notthere" TEMP="" $COMPILE prog.cob], [0], [],
+# Note: on Windows, GCC/MinGW randomly fails when it can't access the temporary directory,
+# so we only perform the COBOL-to-C translation. This is sufficient since the error we want
+# to check is a compile-time error. Note that we nevertheless perform a full compilation
+# afterwards, without tweaking the temporary, so that we can test the actual execution.
+AT_CHECK([TMPDIR="" TMP="notthere" TEMP="" $COMPILE_ONLY prog.cob], [0], [],
 [libcob: warning: Temporary directory TMP is invalid, adjust TMPDIR!
 ])
+AT_CHECK([$COMPILE prog.cob], [0], [], [])
 AT_CHECK([$COBCRUN_DIRECT ./prog], [0], [OK], [])
```

Comment:
The only point of testing the execution was to verify that this does work - without the recompile, because we adjust the broken environment before calling the C compiler.

The goal is good, but if we don't find a way to ensure that cobc's `setenv` reaches the program envrionment of the called C compiler, I'd rather let this fail "from time to time" instead of globally accepting a failure here.

I'd suggest to move this to a draft until someone finds the way to tweak this "correctly" (I _guess_ that means setting some environment variable; it would likely be reasonable to ask the MSYS2 folks (possibly by creating an issue in their tracker - if this happens under MSYS2).